src/engine/renderer/tr_shader.cpp

```diff
@@ -6477,11 +6488,18 @@ void R_ShaderExp_f()
 	}
 
 	const char* buffer_p = buffer.c_str();
+	expression_t exp;
```

Comment:
> It has no risk of getting log suppressed so changing to Str::Format + CommandInteractionMessage is a pure regression in terms of porting to Print.

I want the code to be explicit about what it does. when reading it I don't whant to have to guess the behaviour according to the environment, or to write a comment remembering the guess, as was the solution I chose to please you a previous time you said the same:

https://github.com/DaemonEngine/Daemon/blob/2ec4cb73e145633d3616f0d92781a7b9c24e8ffd/src/engine/sys/sdl_glimp.cpp#L1809-L1813

The written code should not only instruct the compiler what to do but also the reader what it does.

This code is not critical about performance so doing code golf here just looks counter-productive to me.

Reply:
I removed the useless lines but kept the `Log::CommandInteractionMessage` which makes perfect sense since it's a command interaction message.

One thing we may do in the future is to add a `Log::CommandInteractionMessage` that can act like `Log::Notice` and do string building.

Reply:
OK, you got rid of `----` so now it's an improvement :)